src/python/pants/backend/python/target_types.py

```diff
@@ -1349,3 +1350,77 @@ class PythonDistribution(Target):
         See {doc_url('python-distributions')}.
         """
     )
+
+
+class SetuptoolsSCMDummySourceField(OptionalSingleSourceField):
+    """A dummy SourceField for participation in the codegen machinery."""
+
+    alias = "_dummy_source"  # Leading underscore omits the field from help.
+    help = "A version generated by setuptools_scm"
+
+
+class SetuptoolsSCMTagRegexField(StringField):
+    default = None
+    alias = "tag_regex"
+    help = softwrap(
+        """
+        A Python regex string to extract the version part from an SCM tag.
+
+        The regex needs to contain either a single match group, or a group named version,
+        that captures the actual version information.
+
+        Note that this is unrelated to the tags field and to Pants's own tags concept.
+
+        See https://github.com/pypa/setuptools_scm for details.
+        """
+    )
+
+
+class SetuptoolsSCMWriteToField(StringField):
+    required = True
+    alias = "write_to"
```

Comment:
Agreed that might confuse people who expect Pants to save to disk. 

`output_path`?

Reply:
This is tricksy, because usually with codegen we compute an output path based on the input file's name or information in the input file. Here we're making the BUILD file author confront a little piece of the codegen machinery directly...

How about `generate_to` and `generate_template` ?

Reply:
Or just `template`, really.